Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -154,29 +153,58 @@ open class DropboxMobileOAuthManager: DropboxOAuthManager {
         }
     }
     
-    func extractfromDAuthURL(_ url: URL) -> DropboxOAuthResult {
+    func extractfromDAuthURL(_ url: URL, completion: @escaping DropboxOAuthCompletion) {
         switch url.path {
         case "/connect":
-            var results = [String: String]()
-            let pairs  = url.query?.components(separatedBy: "&") ?? []
-            
-            for pair in pairs {
-                let kv = pair.components(separatedBy: "=")
-                results.updateValue(kv[1], forKey: kv[0])
-            }
-            let state = results["state"]?.components(separatedBy: "%3A") ?? []
-            
-            let nonce = UserDefaults.standard.object(forKey: kDBLinkNonce) as? String
-            if state.count == 2 && state[0] == "oauth2" && state[1] == nonce! {
-                let accessToken = results["oauth_token_secret"]!
-                let uid = results["uid"]!
-                return .success(DropboxAccessToken(accessToken: accessToken, uid: uid))
+            let results = OAuthUtils.extractParamsFromUrl(url)
+            if let authSession = authSession {
+                handleCodeFlowResults(results, authSession: authSession, completion: completion)
             } else {
-                return .error(.unknown, "Unable to verify link request")
+                handleTokenFlowResults(results, completion: completion)
             }
         default:
-            return .error(.accessDenied, "User cancelled Dropbox link")
+            completion(.error(.accessDenied, "User cancelled Dropbox link"))
+        }
+    }
+
+    private func handleCodeFlowResults(
+        _ results: [String: String], authSession: AuthSession, completion: @escaping DropboxOAuthCompletion
```

Comment:
results here is pretty bad naming. I dont know what results is in the context of this function. At the very minimum, lets update this name to reflect the fact that its a dictionary. I've made a comment in changing this results type all-together in `OauthUtils`.

Reply:
its type says it's a dictionary though : ) I will think of a better name.